File: src/virgo_agentic_dag/domain/node/node_state.py
"""The controller's own state for a node, independent of its pull request."""

from __future__ import annotations

from enum import Enum


class NodeState(Enum):
    PENDING = "pending"
    IN_PROGRESS = "in_progress"
    RESTING = "resting"
    NEEDS_HUMAN = "needs_human"
    MERGED = "merged"
    ERRORED = "errored"
    SKIPPED = "skipped"

    def is_terminal(self) -> bool:
        """Report whether the controller is finished with this node for good."""
        return self in (
            NodeState.MERGED,
            NodeState.NEEDS_HUMAN,
            NodeState.ERRORED,
            NodeState.SKIPPED,
        )

File: web/entities/nodeState.ts
export type NodeState =
  | 'pending'
  | 'in_progress'
  | 'resting'
  | 'merged'
  | 'needs_human'
  | 'errored'
  | 'skipped';

export const NODE_STATES: readonly NodeState[] = [
  'pending',
  'in_progress',
  'resting',
  'merged',
  'needs_human',
  'errored',
  'skipped',
];
